test/UnitTest/Microsoft.Test.AspNet.OData/Routing/ODataPathRouteConstraintTest.cs

```diff
@@ -67,7 +65,15 @@ public class ODataPathRouteConstraintTest
             { "some(sub)and&other=delims" },
             { "some(delims)but%2Bupper:escaped" },              // "some(delims)but+upper:escaped"
             { "some(delims)but%2blower:escaped" },              // "some(delims)but+lower:escaped"
+#if NETCORE
+            // RFC3986 specifies "[" and "]" and gen-delims in BNF notation in section  2.2
+            // but section 3.2.2 specifies that the are only valid in a Uri as part of an IPV6
+            // host. Therefore, they are not valid in any part of a Uri.
+            { ":@" },                                           // general delimeters that work
```

Comment:
:@  can't work in Classic?

Reply:
":", "@" are not escaped in either version. "[" and "]" are escaped by HttpRequest and it's support classes. They are not escaped by HttpRequestMessage. 

See (PathStringHelper)[https://github.com/aspnet/HttpAbstractions/blob/0da88e0dda36d48ae68ac24289c349605cd94828/src/Microsoft.AspNetCore.Http.Abstractions/Internal/PathStringHelper.cs] for the encoding rules in AspNetCore.

The AspNet code tests all 4 so I'm not sure I understand the question?

#if NETCORE

            // RFC3986 specifies "[" and "]" and gen-delims in BNF notation in section  2.2

            // but section 3.2.2 specifies that the are only valid in a Uri as part of an IPV6

            // host. Therefore, they are not valid in any part of a Uri.

            { ":@" },                                           // general delimeters that work

#else

            // For back-compat, AspNet still tests with "[" and "]" as gen-delims.

            { ":[]@" },                                         // general delimeters that work

#endif